cherrypy/_cplogging.py

```diff
@@ -262,6 +262,8 @@ def access(self):
                  'f': dict.get(inheaders, 'Referer', ''),
                  'a': dict.get(inheaders, 'User-Agent', ''),
                  'o': dict.get(inheaders, 'Host', '-'),
+                 'i': str(uuid.uuid4()),
```

Comment:
Why exactly do you need a new uuid for each access call shouldn't it be per-request?

Reply:
because if you want to set different logformat with cherrypy._cplogging.LogManager.access_log_format(format)
if you want to have unique uuid for each request I thougth that could be the right place to set the uuid.
Use case:
I'm developing a simple semaphore dashboard for our CI; I want that the dashboard logs would be compliant with our best practices: in my case the configuration would be:
cherrypy._cplogging.LogManager.access_log_format('{"request_id":"%(i)s", "user-agent":"%(a)s", "timestamp": "%(z)s", "referrer":"%(f)s", "status":%(s)s, "remote_ip":"%(h)s"}')

Reply:
I mean that it would call `uuid.uuid4()` every time `access()` is called if I remember correctly, meaning that there can be several different UUIDs generated per request if you'd call `access()` more then once. Right?

Reply:
I've explored prior art and found out that @OpenStack uses `%(uuid)s` format string for this, so I'd like to follow the same convention.
Do you know any project using `%(i)s` for this?

Reply:
oh.. okay, makes sense. But I'm still not sure about multiple calls

Reply:
every call should have an unique uuid to track every single request; If there's a better place for doing that, it's even better!

Reply:
Imagine that you can manually invoke `access()` several times in your handler. What would be the result?

Reply:
maybe I didn't understand well the code but isn't it already invoked every time there's an access log?

Reply:
It is invoked once per request, but who stops users from doing this again in their code?

Reply:
I got your point but I don't have idea how to solve this: maybe i should wrap the creation of the uuid like the time() and time_z() methods. Could be reasonable?

Reply:
I'd go for some lazy property: on the first invocation it should save the result to `cherrypy.serving.request` (which is thread-local) and if the value is saved there just return it. I think, you should put a lazy object here, so that it would be called only if rendered into log with `__str__()`.
Also I'm thinking of that not everyone might want to have that in their logs, so maybe in makes sense to only invoke this if they've added `%(i)s` into their format string.

Reply:
(applies to both `%(i)s` and `%(z)s`)

Reply:
Agree, for that reason I added only the atoms without touching the default access_log_format which is in standard Apache format

Reply:
But they are still not lazy, meaning we waste CPU time calculating them.

Reply:
got it, I'll fix it, thanks for the tips and sorry for the poor start quality: like I said it's my first public PR and I'm a sysadmin not a developer :D

Reply:
No worries @spagno, I'm not judging. I'm here to help and walk you through the process :) You're doing great!